core/src/main/java/io/grpc/ChannelImpl.java

```diff
@@ -561,4 +590,34 @@ public Long parseAsciiString(String serialized) {
       return Long.parseLong(valuePart) * factor;
     }
   }
+
+  /**
+   * Should only be called from methods that are themselves synchronized.
+   */
+  private synchronized void releaseDeadlineCancellationExecutor() {
+    if (deadlineCancellationExecutor != null) {
```

Comment:
After removing the call to this from transportTerminated, the `!= null` would no longer be necessary and the method could easy be inlined in `shutdown()`.